src/main/java/org/marketplace/services/AdvertisementImageManagementService.java

```diff
@@ -15,21 +16,34 @@ public AdvertisementImageManagementService(AdvertisementImageManagementRepositor
     }
 
     public AdvertisementImage addImage(AdvertisementImage advertisementImage) {
-        return advertisementImageManagementRepository.save(advertisementImage);
+        try {
+            getAdvertisementImageById(advertisementImage.getId());
+            throw new EntityNotFoundException(String.format("Advertisement image with id: %d already exists!", advertisementImage.getId()));
```

Comment:
hey team, is it some typo? we add image but actually we get image and regardless of the result we fail with an exception..?

Reply:
Hey, as far as I remember it is deliberate. 

What I wanted to do, is to 

1. Check if the image exists (so to see if created Image instance already exists - if so, throw an error.

Okay at this point I see that it actually is a mistake. The error thrown should be `EntityFoundException`

So yes, I will fix it soon.

The second step should be like:
2. If entity was not wound, then proceed saving the image.
3. If entity was found, propagate the error and finish.

Thanks for catching that @dimmonn